src/features/brain-atlas-viewer/context.tsx

```diff
@@ -0,0 +1,74 @@
+import { atomFamily } from 'jotai/utils';
+import { atom } from 'jotai';
+
+import isNil from 'lodash/isNil';
+import { downloadAsset } from '@/api/entitycore/queries/assets';
+import {
+  getBrainAtlasRegion,
+  getBrainAtlasRegions,
+} from '@/api/entitycore/queries/general/brain-atlas';
+import { arrayBufferToString } from '@/utils/buffer';
+import { tryCatch } from '@/api/utils';
+import { env } from '@/env';
+
+export async function resolveBrainRegionAtlasMesh({
+  atlasRegionId,
+  atlasId = env.NEXT_PUBLIC_DEFAULT_BRAIN_ATLAS_ID,
+}: {
+  atlasRegionId: string;
+  atlasId?: string;
+}) {
+  const { data: atlas, error: atlasError } = await tryCatch(
+    getBrainAtlasRegion({ atlasId, atlasRegionId })
+  );
+  if (atlasError)
+    throw Error(
+      `Unable to retrieve data for brain region id "${atlasRegionId}" in atlas "${atlasId}`
+    );
+  const atlasAssetId = atlas.assets.at(0)?.id;
+  if (isNil(atlasAssetId))
+    throw Error(`No mesh data available for brain region ID "${atlasRegionId}`);
+  const { data: asset, error: assetError } = await tryCatch(
+    downloadAsset<ArrayBuffer>({
+      // @ts-expect-error
+      entityType: 'brain-atlas-region',
+      entityId: atlas.id,
+      asRawResponse: false,
+      id: atlasAssetId,
+    })
+  );
+
+  if (assetError)
+    throw Error(
+      `Failed to download mesh asset (ID: "${atlasAssetId}") for brain region ID "${atlasRegionId}`
+    );
+  const data = arrayBufferToString(asset);
```

Comment:
Conceptually more correct way to do that is in downloadAsset function so that res.text() is called under the hood which relies on response headers to determine encoding; otherwise we assume that it's utf-8.

Maybe we add a note to address later? But that's very minor.